web/src/containers/QuickExchange/index.tsx

```diff
@@ -0,0 +1,373 @@
+    const walletQuote = React.useMemo(() => getWallet(quote.currency, wallets), [quote.currency, wallets]);
+
+    const walletsBaseList = walletsBaseUnit.length ? walletsBaseUnit.map(item => item.currency && item.currency.toUpperCase()) : [];
+    const walletsQuoteList = walletsQuoteUnit.length ? walletsQuoteUnit.map(item => item.currency && item.currency.toUpperCase()) : [];
+
+    const handleChangeValue = (value: string, key: string) => {
+        const precision = key === 'base_value' ? basePrecision : quotePrecision;
+
+        if (walletBase && value.match(precisionRegExp(precision))) {
+            const [baseAmount, quoteAmount] = key === 'base_value'
+                ? getBaseAmount(walletBase, value, marketPrice.price, base.amount, quote.amount)
+                : getQuoteAmount(walletBase, value, marketPrice.price, base.amount, quote.amount);
+
+            const newBaseValue = {
+                ...base,
+                amount: baseAmount,
+            };
+            const newQuoteValue = {
+                ...quote,
+                amount: quoteAmount,
+            };
+            setBaseData(newBaseValue);
+            setQuoteData(newQuoteValue);
+        }
+    };
+
+    const handleChangeDropdownBase = React.useCallback((index: number) => {
+        const market = { id: '' };
+        const value = walletsBaseList[index];
+        const marketExists = markets.some(item => item.id === (value + quote.currency).toLowerCase());
+        const marketMirrorExists = markets.some(item => item.id === (quote.currency + value).toLowerCase());
+
+        if (marketExists) {
+            market.id = (value + quote.currency).toLowerCase();
+            const currentMarket = getMarket(market.id, markets);
+
+            if (currentMarket) {
+                setType('buy');
+                setMarket(market.id);
+                setBasePrecision(currentMarket.amount_precision);
+                setQuotePrecision(currentMarket.price_precision);
+            }
+        } else if (marketMirrorExists) {
+            market.id = (quote.currency + value).toLowerCase();
+            const currentMarket = getMarket(market.id, markets);
+
+            if (currentMarket) {
+                setType('sell');
+                setMarket(market.id);
+                setBasePrecision(currentMarket.price_precision);
+                setQuotePrecision(currentMarket.amount_precision);
+            }
+        }
+
+        if (base.currency !== value && base.currency !== '') {
+            setQuoteData(DEFAULT_VALUE);
+        }
+
+        const newBaseCurrency: QuickExchangeInterface = {
+            amount: '',
+            currency: value,
+        };
+        setBaseData(newBaseCurrency);
+        setDropdownKey('base_unit');
+    }, [walletsBaseList, markets, base, quote]);
+
+    const handleChangeDropdownQuote = React.useCallback((index: number) => {
```

Comment:
Remove duplicating functional. There is no need to have 2 almost similar functions `handleChangeDropdownBase` and `handleChangeDropdownQuote`. Pass `side` as parameter.